cypress/support/gutenberg/commands/assertions.js

```diff
@@ -184,44 +297,65 @@ export function assertClassName( subject, customSelector = '', expectedValue = '
 		assertBackend = true,
 		assertFrontend = true,
 		delay = 0,
+		afterFrontendAssert = () => {},
+		afterBackendAssert = () => {},
 	} = options
 
 	cy.wp().then( wp => {
 		cy.publish()
 		cy.wait( delay )
+		const blockPath = getBlockStringPath( wp.data.select( 'core/block-editor' ).getBlocks(), subject.data( 'block' ) )
 
-		const block = wp.data.select( 'core/block-editor' ).getBlock( subject.data( 'block' ) )
-		const saveElement = createElementFromHTMLString( wp.blocks.getBlockContent( block ) )
-		const parsedClassList = Array.from( saveElement.classList ).map( _class => `.${ _class }` ).join( '' )
-
-		cy
-			.get( subject )
-			.then( $block => {
-				// Assert editor classes.
-				if ( assertBackend ) {
-					assert.isTrue(
-						!! $block.find( `${ customSelector }.${ expectedValue }` ).length,
-						`${ expectedValue } class must be present in ${ customSelector } in Editor`
-					)
-				}
-
-				// Assert frontend classes.
-				// Check if we're asserting the parent element.
-				if ( assertFrontend ) {
-					if ( parsedClassList.match( customSelector ) ) {
-						assert.isTrue(
-							!! parsedClassList.match( expectedValue ),
-							`${ expectedValue } class must be present in ${ customSelector } in Editor`
-						)
-					} else {
-						// Otherwise, search the element
+		cy.getBlockAttributes().then( attributes => {
+			const selector = `.${ attributes.className }`
+			cy
+				.get( subject )
+				.then( $block => {
+					// Assert editor classes.
+					if ( assertBackend ) {
 						assert.isTrue(
-							!! Array.from( saveElement.querySelector( customSelector ).classList ).includes( expectedValue ),
+							!! $block.find( `${ customSelector }.${ expectedValue }` ).length,
 							`${ expectedValue } class must be present in ${ customSelector } in Editor`
 						)
+						afterBackendAssert()
 					}
-				}
-			} )
+
+					// Assert frontend classes
+					if ( assertFrontend ) {
+						cy.getPostUrls().then( ( { editorUrl, previewUrl } ) => {
+							cy.visit( previewUrl )
+							cy.document().then( doc => {
+								const element = doc.querySelector( selector )
+								if ( element ) {
+									cy.get( selector ).invoke( 'attr', 'class' ).then( classNames => {
```

Comment:
We can further simplify this. `Array.from( element.classList )` returns the classes already